Pocket/pocket-ios
# ci(swiftlint): updated rules

## Summary
Update our SwiftLint rules:
- redundant_string_enum_value
- file_header
- overridden_super_call (address issues in tests)

Fix some SwiftLint configurations by moving appropriate rules to `analyzer_rules`

## References 
N / A 

## Implementation Details
* Add two rules `redundant_string_enum_value` and `file_header` see warnings addressed in separate commits
* Address issues with `overridden_super_call` to match closely with Firefox's SwiftLint file
* Fix SwiftLint configurations and address remaining warnings after upgrading SwiftLint versions

## Test Steps
Build and run app as expected. All issues should be addressed and project should work normally.

## PR Checklist:
- N / A Added Unit / UI tests

## Screenshots
